packages/nuxt/src/app/composables/fetch.ts

```diff
@@ -140,12 +140,9 @@ export function useFetch<
     controller?.abort?.()
     controller = typeof AbortController !== 'undefined' ? new AbortController() : {} as AbortController
 
-    const isLocalFetch = typeof _request.value === 'string' && _request.value.startsWith('/')
-    let _$fetch = opts.$fetch || globalThis.$fetch
+    const isLocalFetch = typeof _request.value === 'string' && _request.value.startsWith('/') && (!unref(opts.baseURL) || unref(opts.baseURL)!.startsWith('/'))
     // Use fetch with request context and headers for server direct API calls
-    if (import.meta.server && !opts.$fetch && isLocalFetch) {
-      _$fetch = useRequestFetch()
-    }
+    const _$fetch = opts.$fetch || ((import.meta.server && isLocalFetch) ? useRequestFetch() : globalThis.$fetch)
```

Comment:
Small perf: I'm not sure if this would be tree-shaken without modification so worth to double check if you can quickly check. Seems we need `isLocalFetch` computation only in server and without explicit `$fetch` option.

Reply:
yes, it's still tree-shaken.